### PR TITLE
Custom animations

### PR DESCRIPTION
**Pull request by albertogasparin**
Allow registering custom Velocity animations and allow them to be defined even after the angular-velocity script inclusion.

### CODE REVIEW

**Comment by cgwyllie** [comment]
Hey @albertogasparin, this looks great! Definitely a useful feature =D

Can you briefly explain what the `Velocity.Redirects` collection is, and if this still contains all 'packaged' effects?

Thanks for the contribution, I'll aim to get it packaged up tomorrow evening.

File: angular-velocity.js [hunk 1]
```diff
@@ -252,24 +250,29 @@
...
-			app.animation(leaveSelector, makeLeaveAnimationFor(animation));
-		}
-	});
+	angular
+		.module('angular-velocity', ['ngAnimate'])
+		.config(['$animateProvider', function ($animateProvider) {
+			// Use the factories to define animations for all velocity's sequences
+			angular.forEach(Velocity.Redirects, function (_, animation) {
+				var IN = 'In',
+					OUT = 'Out',
```
Comment (cgwyllie) [inline angular-velocity.js]: Could you fix the indentation before merge? I can do it before minification if you'd rather that, just let me know.

Comment (albertogasparin) [inline angular-velocity.js]: Sorry but what do you mean by "fix the indentation"? I've tried to keep your settings. So in my editor with tab indentation set to 4 everything is correctly aligned. It looks odd here in Github by the way.

**Comment by albertogasparin** [comment]
`Velocity.RegisterEffect()` is the official API that can be used to add custom animations. It is the same API used by [the effect pack](https://github.com/julianshapiro/velocity/blob/master/velocity.ui.js#L713-L715).

By looking a what that API does, I have found that it [assigns the animation as a property](https://github.com/julianshapiro/velocity/blob/master/velocity.ui.js#L97) on the `Velocity.Redirects` object. So that object is what we should use as a collection for the `forEach` loop because it contains every animation registered.

File: angular-velocity.js (hunk 1 above)
Comment (cgwyllie) [inline angular-velocity.js]: Ah, sorry, yep it's the online preview being a little messed up :)

**Comment by cgwyllie** [comment]
@albertogasparin Brilliant, that seems like exactly the property to iterate, thanks!

**Pull request merged**